File: src/cmd/start.go
package cmd

import (
	"fmt"

	"angel/src/core"
	"angel/src/core/launchctl"

	"github.com/alecthomas/kong"
)

type StartCmd struct {
	Name string `arg:"" help:"Service name to start."`
}

func (s *StartCmd) Run(angel *core.Angel, ctx *kong.Context) error {
	return angel.WithMatch(s.Name, false, ctx, func(daemon core.Daemon) error {
		output, err := launchctl.KickstartKill(daemon)
		if err != nil {
			return err
		}
		fmt.Print(string(output))
		fmt.Printf("started %s\n", daemon.Name)
		return nil
	})
}

File: src/commands/start.rs
use crate::angel::Angel;
use crate::cli::StartArgs;
use crate::error::Result;
use crate::launchctl;
use crate::output::stdout;

pub fn run(angel: &Angel, args: &StartArgs) -> Result<()> {
    let daemon = angel.daemons.get_match(&args.name, args.exact)?;
    let result = match args.kill {
        true => launchctl::kickstart_kill(daemon)?,
        false => launchctl::kickstart(daemon)?,
    };
    stdout::write(&result.output);
    match result.success() {
        true => stdout::success(&format!("started {}", daemon.name)),
        false => stdout::error(&format!(
            "failed to start {}: {}",
            daemon.name, result.stderr
        )),
    }
    Ok(())
}
